File: T2_2_81/template/retrieve_sp.m
function  arr=retrieve_sp(sparsearr)
        arr = zeros(sparsearr.m, sparsearr.n);
        for j = 1:sparsearr.n
                index = (sparsearr.colptr(j) : (sparsearr.colptr(j+1) - 1));
                arr(sparsearr.rowval(index), j) = sparsearr.nzval(index);
        end
end

File: basic_network/2/retrieve_sp.jl
using SparseArrays

function retrieve_sp(sparsearr::Any)
    arr = zeros(sparsearr.m, sparsearr.n)
    for j = 1:sparsearr.n
        index = (sparsearr.colptr[j]:(sparsearr.colptr[j+1]-1))
        arr[sparsearr.rowval[index], j] = sparsearr.nzval[index]
    end
    return arr
end

function retrieve_sp(sparsearr::Dict)
    arr = zeros(sparsearr["m"], sparsearr["n"])
    for j = 1:sparsearr["n"]
        index = (sparsearr["colptr"][j]:(sparsearr["colptr"][j+1]-1))
        arr[sparsearr["rowval"][index], j] = sparsearr["nzval"][index]
    end
    return arr
end
